src/codemodder/dependency_management/pyproject_writer.py

```diff
@@ -0,0 +1,55 @@
+import tomlkit
+from typing import Optional
+from copy import deepcopy
+from codemodder.dependency import Dependency
+from codemodder.change import Action, Change, ChangeSet, PackageAction, Result
+from codemodder.dependency_management.base_dependency_writer import DependencyWriter
+from codemodder.diff import create_diff_and_linenums
+
+
+class PyprojectWriter(DependencyWriter):
+    def add_to_file(
+        self, dependencies: list[Dependency], dry_run: bool = False
+    ) -> Optional[ChangeSet]:
+        pyproject = self._parse_file()
+        original = deepcopy(pyproject)
+
+        try:
+            pyproject["project"]["dependencies"].extend(
+                [f"{dep.requirement}" for dep in dependencies]
+            )
+        except tomlkit.exceptions.NonExistentKey:
+            return None
```

Comment:
I think we probably ought to have some kind of debug logging either here or at the caller.